.github/workflows/release.yaml

```diff
@@ -0,0 +1,73 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.*'
+
+jobs:
+  build:
+    name: Build Binaries
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        goos: [linux, windows, darwin]
+        goarch: [amd64, arm64]
+        wasm-tools-version: ["1.216.0"]
+        exclude:
+          - goos: windows
+            goarch: arm64
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Set up wasm-tools
+        uses: bytecodealliance/actions/wasm-tools/setup@v1
+        with:
+          version: ${{ matrix.wasm-tools-version }}
+
+      - name: Run Go tests
+        run: go test -v ./...
+
+      - name: Build Binary
+        run: |
+          mkdir -p build
+          GOOS=${{ matrix.goos }} GOARCH=${{ matrix.goarch }} go build -o build/wit-bindgen-go_${{ matrix.goos }}_${{ matrix.goarch }} ./cmd/wit-bindgen-go
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: wit-bindgen-go_${{ matrix.goos }}_${{ matrix.goarch }}
+          path: build/wit-bindgen-go_${{ matrix.goos }}_${{ matrix.goarch }}
```

Comment:
Does the name matter on final download?

Can this be a subdirectory eg builds/amd64-windows/wit-binden-go?

Reply:
The binary in the release page will have the name specified here.

Reply:
> Can this be a subdirectory eg builds/amd64-windows/wit-binden-go?

I believe so.

Reply:
OK, I think this should build each binary as `wit-bindgen-go`, and then create a `.tgz` file in `dist` directory named `wit-bindgen-go-os-arch.tgz`.

Reply:
Done